src/mds/MDCache.cc

```diff
@@ -8315,9 +8315,37 @@ int MDCache::path_traverse(MDRequestRef& mdr, MDSContextFactory& cf,
   if (flags & MDS_TRAVERSE_CHECK_LOCKCACHE)
     mds->locker->find_and_attach_lock_cache(mdr, cur);
 
-  if (mdr && mdr->lock_cache) {
-    if (flags & MDS_TRAVERSE_WANT_DIRLAYOUT)
+  if (mdr) {
+    if (mdr->lock_cache && (flags & MDS_TRAVERSE_WANT_DIRLAYOUT))
       mdr->dir_layout = mdr->lock_cache->get_dir_layout();
+    const cref_t<MClientRequest> &req = mdr->client_request;
+    Session *session = mds->get_session(req);
+    if (session && !session->info.has_feature(CEPHFS_FEATURE_ALTERNATE_NAME) &&
+	req->get_op() != CEPH_MDS_OP_LOOKUP &&
+	req->get_op() != CEPH_MDS_OP_LOOKUPHASH &&
+	req->get_op() != CEPH_MDS_OP_LOOKUPPARENT &&
+	req->get_op() != CEPH_MDS_OP_LOOKUPINO &&
+	req->get_op() != CEPH_MDS_OP_LOOKUPNAME &&
+	req->get_op() != CEPH_MDS_OP_LOOKUPSNAP &&
+	req->get_op() != CEPH_MDS_OP_RMSNAP &&
+	req->get_op() != CEPH_MDS_OP_LSSNAP &&
+	req->get_op() != CEPH_MDS_OP_GETATTR &&
+	req->get_op() != CEPH_MDS_OP_READDIR &&
+	req->get_op() != CEPH_MDS_OP_UNLINK &&
+	req->get_op() != CEPH_MDS_OP_RMDIR) {
+      MutationImpl::LockOpVec lov;
+      /* We need 'As' caps for the fscrypt context */
+      lov.add_rdlock(&cur->authlock);
```

Comment:
Sorry for late, I was on PTO last week.

This will work IMO, but as I mentioned this will break the Locker order.

Could we do this in the following code ?

Reply:
@luis-henrix ping?